indexer/osm_editor.cpp

```diff
@@ -138,7 +138,9 @@ namespace osm
 // (e.g. insert/remove spaces after ';' delimeter);
 
 Editor::Editor()
-  : m_notes(editor::Notes::MakeNotes())
+  : m_config(make_shared<editor::EditorConfig>())
```

Comment:
I dislike the idea to wrap all usage of m_config to atomic_load/atomic_store. I suggest you do wrap it in a light-weight wrapper, like:
```c++
class ConfigStorage
{
  public:
    ConfigStorage() = default;

    void Set(shared_ptr<editor::EditorConfig> config) { atomic_store(...); }
    shared_ptr<editor::EditorConfig> Get() { atomic_load(...); }

  private:
    shared_ptr<editor::EditorConfig> m_config = make_shared<editor::EditorConfig>();
};
```
and use it instead of raw shared_ptr.